UsbAccess/UsbAccess.cpp

```diff
@@ -508,6 +563,172 @@ namespace Plugin {
         returnResponse(result);
     }
 
+    uint32_t UsbAccess::getPartitionCountWrapper(const JsonObject &parameters, JsonObject &response)
+    {
+        LOGINFOMETHOD();
+        bool result = true;
+        char buffer[32] = "";
+        string command =  "lsblk -o NAME,MOUNTPOINT | grep \"sd[a-z]\" | grep \"/\" |wc -l";
+        FILE* fp = popen(command.c_str(), "r");
+        if (fp == NULL) 
+        {
+            result = false;
+        }   
+        
+        while (fgets(buffer, 32, fp) != NULL);
+
+        string PartitionCount = buffer;
+
+        int ret = pclose(fp);
+        if (ret == -1) 
+        {
+            result = false;
+        }
+        response["PartitionCount"] = PartitionCount;
+        returnResponse(result);
+    }
+
+
+
+    uint32_t UsbAccess::getPartitionInfoWrapper(const JsonObject &parameters, JsonObject &response)
+    {
+        LOGINFOMETHOD();
+        bool result = true;
+        char buffer[32] = "";
+        unsigned long long int blocksize = 0;
+        JsonArray arr;
+        JsonObject ent;
+        string pathParam;
+        if (parameters.HasLabel("path"))
+            pathParam = parameters["path"].String();
+
+        struct statvfs fs;
+        if (statvfs(pathParam.c_str(), &fs) == -1) {
+            result = false;
+        }
+
+        string command_str1 = "df -T  ";
+        string command_str2 = " | tail -n +2 | awk '{print $2}'";
+        string command = command_str1 + pathParam + command_str2;
+        FILE* fp = popen(command.c_str(), "r");
+        if (fp == NULL) 
+        {
+            result = false;
+        }   
+        
+        while (fgets(buffer, 32, fp) != NULL);
```

Comment:
Same as above comment. fp is used in logic event after check with NULL. This could cause crash.